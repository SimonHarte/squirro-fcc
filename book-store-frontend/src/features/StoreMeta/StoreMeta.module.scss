@import "../../styles/variables.scss";

.divider {
  display: inline-block;
  margin: 0 ($spacing-base * 2);
}

.link {
  border-bottom: 1px solid transparent;
  transition: border-color 0.3s;

  &:focus,
  &:hover {
    border-color: inherit;
  }
}
