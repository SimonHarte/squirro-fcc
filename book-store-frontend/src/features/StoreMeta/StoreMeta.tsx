import { type ReactElement, type FC } from 'react';
import formatDate from '../../libs/formatDate';
import type { StoreMetaProps } from './StoreMeta.types';
import styles from './StoreMeta.module.scss';

const StoreMeta: FC<StoreMetaProps> = ({
  date,
  website,
}: StoreMetaProps): ReactElement => {
  const displayDate = formatDate(date);

  return (
    <div>
      <time dateTime={date}>{displayDate}</time>
      <span className={styles.divider}>-</span>
      <a href={website}>{website}</a>
    </div>
  );
};

export default StoreMeta;
