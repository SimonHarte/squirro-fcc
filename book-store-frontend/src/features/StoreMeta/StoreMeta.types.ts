export type StoreMetaProps = {
  date: string;
  website: string;
};
