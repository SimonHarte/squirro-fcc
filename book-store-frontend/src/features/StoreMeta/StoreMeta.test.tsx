import { shallow } from 'enzyme';
import StoreMeta from './index';

describe('StoreMeta', () => {
  test('should render correctly', () => {
    const wrapper = shallow(
      <StoreMeta
        date={new Date(1990, 4, 1).toISOString()}
        website="http://test-site.com"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
