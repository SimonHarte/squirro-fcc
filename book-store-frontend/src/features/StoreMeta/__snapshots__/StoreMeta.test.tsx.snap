// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreMeta should render correctly 1`] = `
<div>
  <time
    dateTime="1990-04-30T22:00:00.000Z"
  >
    01.05.1990
  </time>
  <span
    className="divider"
  >
    -
  </span>
  <a
    className="link"
    href="http://test-site.com"
  >
    http://test-site.com
  </a>
</div>
`;
